include/swift/Frontend/OutputFileMap.h

```diff
@@ -74,19 +73,16 @@ class OutputFileMap {
   /// Dump the OutputFileMap to the given \p os.
   void dump(llvm::raw_ostream &os, bool Sort = false) const;
 
+  /// Write the OutputFilemap for the \p inputs so it can be parsed.
+  void write(llvm::raw_ostream &os, ArrayRef<StringRef> inputs) const;
+
 private:
-  /// \brief Parses the given \p Buffer into the OutputFileMap, taking ownership
-  /// of \p Buffer in the process.
-  ///
-  /// When non-empty, \p workingDirectory is used to resolve relative paths in
-  /// the output file map.
-  ///
-  /// \returns true on error, false on success
-  bool parse(std::unique_ptr<llvm::MemoryBuffer> Buffer,
-             StringRef workingDirectory);
+  /// \brief Parses the given \p Buffer and returns either an OutputFileMap or
+  /// error, taking ownership of \p Buffer in the process.
```

Comment:
Huh, it looks like nothing actually needs this to take ownership of the buffer. Maybe that can go back to being a reference (later).

Reply:
By "later", I'm guessing you mean "after this PR".